service/src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -268,6 +264,40 @@ public boolean updateWorkspace(
     return updated;
   }
 
+  /** Update a workspace properties */
+  @WriteTransaction
+  public boolean updateWorkspaceProperties(UUID workspaceUuid, Map<String, String> propertyMap) {
+    // get current property in this workspace id
+    String selectPropertiesSql = "SELECT properties FROM workspace WHERE workspace_id = :id";
+    MapSqlParameterSource propertiesParams =
+        new MapSqlParameterSource().addValue("id", workspaceUuid.toString());
+    int rowsAffected;
+    try {
+      String result =
+          jdbcTemplate.queryForObject(selectPropertiesSql, propertiesParams, String.class);
+      logger.info("Retrieved workspace record {}", result);
+      Map<String, String> properties = DbSerDes.jsonToProperties(result);
+      properties.putAll(propertyMap);
+      final String sql =
+          "UPDATE workspace SET properties = cast(:properties AS jsonb) WHERE workspace_id = :id";
+
+      var params = new MapSqlParameterSource();
+      params
+          .addValue("properties", DbSerDes.propertiesToJson(properties))
+          .addValue("id", workspaceUuid.toString());
+
+      rowsAffected = jdbcTemplate.update(sql, params);
+    } catch (EmptyResultDataAccessException e) {
+      throw new WorkspaceNotFoundException(String.format("Workspace %s not found.", workspaceUuid));
```

Comment:
Hi yu. This sql is for select the properties information in the specific UUID. The property attribute must exist even the value is null. The SQL can still return the null. But the only try catch fail will happens while the workspace UUID is not exist. From my understanding, this will still be workspaceNotFoundException.

Reply:
i see. thanks for explaining